src/lavinmq/mqtt/client.cr

```diff
@@ -0,0 +1,256 @@
+require "openssl"
+require "socket"
+require "../client"
+require "../error"
+require "./session"
+
+module LavinMQ
+  module MQTT
+    class MqttConsumer < LavinMQ::Client::Channel::Consumer
+      getter unacked = 0_u32
+      getter tag : String = "mqtt"
+      property prefetch_count = 1
+
+      def initialize(@client : Client, @queue : Queue)
+        @has_capacity.try_send? true
+        spawn deliver_loop, name: "Consumer deliver loop", same_thread: true
+      end
+
+      private def deliver_loop
+        queue = @queue
+        i = 0
+        loop do
+          queue.consume_get(self) do |env|
+            deliver(env.message, env.segment_position, env.redelivered)
+          end
+          Fiber.yield if (i &+= 1) % 32768 == 0
+        end
```

Comment:
I don't think `consume_get` is blocking if no message is available, so this loop will probably put some stress on the CPU? You most likely need the "wait queue ready" thing.